core/platforms/packaging/bareos.spec

```diff
@@ -1633,32 +1640,53 @@ getent group %1 > /dev/null || groupadd -r %1 \
 getent passwd %1 > /dev/null || useradd -r --comment "%1" --home %{working_dir} -g %{daemon_group} --shell /bin/false %1 \
 %nil
 
+# NOTE: rpm macro with parameter. Rest of line is ignored (taken as parameter).
+# usage:
+# if %%rpm_version_lt <version1> <version2>
+# then
+#   <do something>
+# fi
+%define rpm_version_lt() \
+[ "%1" ] && [ "%2" ] && [ "$(printf "%1\\n%2\\n" | sort --version-sort | head -n 1)" = "%1" ] \
+%nil
```

Comment:
to my surprise `--version-sort` actually handles tilde-versions (i.e. `25.0.0~pre123.xyz` correctly.